### DISCUSSION TITLE
conversion from C# data types?

### DISCUSSION THREAD

**Discussion by sfre-coop** [microsoft/typespec#3105] [General]
I'm trying to spec up a schema which a third party generates reflectively from .Net types. They've sent me the C# which gives me some context and I'd really like not to have to do the conversion by hand.

Is there any tooling for getting from C# to typespec?

Thanks all

**Comment by timotheeguerin** [comment]
We do not have a csharp converter today, I think it is also unlikely we’ll have one in the near future.
We do have however plan to make and openapi3/3.1 converter so at that point you could convert csharp to openapi to typespec  but nothing you can do today

**Reply by sfre-coop** [reply]
Thanks. I might try doing something with the .Net serialiser.

**Reply by bterlson** [reply]
That would be awesome, keep us posted!

**Reply by sfre-coop** [reply]
You asked, here it is: https://github.com/sf105/CSharpToTypeSpec 

Pretty minimal start, but I'm open to contributions.
